web/app/webpack.config.js

```diff
@@ -0,0 +1,107 @@
+const path = require('path');
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+
+// Init/require webpack configs:
+const Env = require('./webpack/env.config');
+
+module.exports = (env, argv) => {
+  const isDevelopment = argv.mode === 'development';
+  const isDevServer = process.env?.BUILD !== '1';
+
+  return {
+    target: 'web',
+    entry: './src/App.js',
+    output: {
+      filename: isDevelopment ? '[name].js' : '[name].[contenthash].js',
+      path: path.resolve(__dirname, 'build'),
+      clean: true,
+      assetModuleFilename: '[path]/[name][ext]'
+    },
+    optimization: {
+      // https://webpack.js.org/guides/caching/#extracting-boilerplate
+      runtimeChunk: 'single',
+      splitChunks: {
+        cacheGroups: {
+          vendor: {
+            test: /[\\/]node_modules[\\/]/,
+            name: 'vendors',
+            chunks: 'all',
+          },
+        },
+      },
+    },
+    module: {
+      rules: [
+        {
+          test: /\.(js|tsx?)$/,
+          exclude: /node_modules/,
+          use: {
+            loader: 'babel-loader',
+            options: {
+              presets: [
+                '@babel/preset-env',
+                '@babel/preset-react',
+                '@babel/preset-typescript'
+              ],
+              plugins: [
+                isDevServer && require.resolve('react-refresh/babel'),
+                // Required for async/await:
+                ['@babel/transform-runtime', { regenerator: true }],
+                // Optimize lodash modules:
+                // https://github.com/lodash/babel-plugin-lodash
+                ['lodash'],
+              ].filter(Boolean),
+            }
+          }
+        },
+        {
+          test: /\.svg$/,
+          exclude: /node_modules/,
+          type: 'asset/resource',
+          use: 'svgo-loader',
+        },
+        {
+          test: /\.(jpeg|jpg|png|woff|woff2|eot|ttf)(\?.*$|$)/,
+          exclude: /node_modules/,
+          type: 'asset',
+        },
+        {
+          // https://www.npmjs.com/package/sass-loader
+          test: /\.s[ac]ss$/i,
+          exclude: /node_modules/,
+          use: [
+            "style-loader",
+            "css-loader",
+            "sass-loader",
+          ],
+        },
+      ]
+    },
+    plugins: [
+      new HtmlWebpackPlugin({
+        template: path.resolve(__dirname, 'public/index.html'),
+        publicPath: '/',
+        alwaysWriteToDisk: true, // See below
```

Comment:
is referencing:

```js

      // Forces the index.html generation, even when using webpack-dev-server
      // https://github.com/jantimon/html-webpack-harddisk-plugin
      isDevServer && new HtmlWebpackHarddiskPlugin(),
```

which is absent for now, as well as the proper config about React Hot Module Reloading with `ReactRefreshWebpackPlugin`